src/tr2/decomp/flares.c

```diff
@@ -208,22 +208,23 @@ void Flare_Create(const bool thrown)
 
 void Flare_SetArm(const int32_t frame)
 {
-    int16_t anim_base;
-
+    int16_t anim_idx;
     if (frame < LF_FL_THROW) {
-        anim_base = g_Objects[O_LARA_FLARE].anim_idx;
+        anim_idx = 0;
     } else if (frame < LF_FL_DRAW) {
-        anim_base = g_Objects[O_LARA_FLARE].anim_idx + 1;
+        anim_idx = 1;
     } else if (frame < LF_FL_IGNITE) {
-        anim_base = g_Objects[O_LARA_FLARE].anim_idx + 2;
+        anim_idx = 2;
     } else if (frame < LF_FL_2_HOLD) {
-        anim_base = g_Objects[O_LARA_FLARE].anim_idx + 3;
+        anim_idx = 3;
     } else {
-        anim_base = g_Objects[O_LARA_FLARE].anim_idx + 4;
+        anim_idx = 4;
```

Comment:
Is it worth adding enums for 1-4 for flares as well as pistols? I guess the frame enums in the if test kinda tell you what these `anim_idx` are setting.

Reply:
I think we should. I will follow it up separately though if we're OK with that as I've just realised I didn't make use of `Anim_TestAbsFrameRange` in `Gun_Pistols_SetArmInfo` in #2224. Makes sense to do these together.

Reply:
Agreed and it's OK to do this in a follow-up PR